routing/missioncontrol.go

```diff
@@ -299,6 +305,36 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 	return bandwidthHints, nil
 }
 
+// updateEdgePolicy updates the channel edge policy parameters if both channel
+// ID and direction match those of the edge locator.
+func updateEdgePolicy(el *edgeLocator, up *lnwire.ChannelUpdate,
+	ep *channeldb.ChannelEdgePolicy) {
+
+	if ep.ChannelID == el.channelID &&
+		uint8(ep.Flags&lnwire.ChanUpdateDirection) == el.direction {
```

Comment:
This is a bit brittle as it assumes that the direction flag is the last bit (this is also true elsewhere in the codebase, but better not to keep assuming it) :)

Instead you can do `(ep.Flags&lnwire.ChanUpdateDirection != 0) == (el.direction != 0)` or similar.

Reply:
Its strange that go doesn't have casting to bool in a simple way. Anyhow, I extracted it into the most straightforward help function, as to me this casting to bool is not self explanatory.